src/core/thread/mle_router.hpp

```diff
@@ -317,6 +317,17 @@ class MleRouter: public Mle
     Child *GetChildren(uint8_t *aNumChildren);
 
     /**
+     * This method sets the max children allowed value for this Thread interface.
+     *
+     * @param[in]  aMaxChildren  The max children allowed value.
+     *
+     * @retval  kThreadErrorNone           Successfully set the max.
+     * @retval  kThreadError_InvalidArgs   If @p aMaxChildren is not in the range [1, kMaxChildren].
+     *
+     */
+    ThreadError SetMaxAllowedChildren(uint8_t aMaxChildren);
```

Comment:
Works for me.  Would be good to document that along with the `kThreadError_InvalidState` return value.